src/Infrastructure/BotSharp.Core/Templating/TemplateRender.cs

```diff
@@ -68,13 +72,46 @@ public void Register(Type type)
             var props = type.GetProperties();
             foreach (var prop in props)
             {
-                Register(prop.PropertyType);
+                RegisterType(prop.PropertyType);
             }
         }
     }
 
 
     #region Private methods
+    private static async ValueTask<Completion> RenderIdentifierTag(string identifier, TextWriter writer, TextEncoder encoder, TemplateContext context)
+    {
+        try
+        {
+            var value = await context.Model.GetValueAsync(TemplateRenderConstant.RENDER_AGENT, context);
```

Comment:
It should print an error message if `TemplateRenderConstant.RENDER_AGENT` is not found in `context`.